disnake/ext/commands/core.py

```diff
@@ -1850,6 +1853,168 @@ async def predicate(ctx: AnyContext) -> bool:
     return check(predicate)
 
 
+def app_check(predicate: AppCheck) -> Callable[[T], T]:
+    """A decorator that adds a check to the :class:`disnake.ext.commands.InvokableApplicationCommand` or its
```

Comment:
Would also be cool to have a backreference from `check` to `app_check`, to direct users into the right direction if they're using app commands and get typing warnings.

Reply:
d6da2cee1ed82bce63efd10cc90470c0991b3bb3